person.rb

```diff
@@ -1,24 +1,29 @@
-require './nameable'
-require './decorator'
+require './corrector'
+require './rental'
 
-class Person < Nameable
-  def initialize(age, name = 'Unknown', parrent_permission: true)
-    super()
-    @id = Random.rand(1..1000)
+class Person
+  attr_accessor :name, :age, :rentals
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1...1000)
     @name = name
     @age = age
-    @parrent_permission = parrent_permission
+    @parent_permission = parent_permission
+    @corrector = Corrector.new
+    @rentals = []
   end
 
-  attr_accessor :name, :age
-  attr_reader :id
-
   def can_use_services?
-    is_of_age? || @parent_permission
+    of_age? || @parent_permission ? true : false
+  end
+
+  def validate_name
+    @name = @corrector.correct_name(@name)
   end
 
-  def correct_name
-    @name
+  def add_rentals(book, date)
```

Comment:
- [x] [OPTIONAL] in terms of naming this function should not be pluralized. the method name should always define what the method does. this method is only adding *one* rental.
```suggestion
  def add_rental(book, date)
```